tools/obscuroscan/obscuroscan_test.go

```diff
@@ -0,0 +1,87 @@
+package obscuroscan
+
+import (
+	"encoding/json"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/obscuronet/obscuro-playground/go/ethclient/mgmtcontractlib"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+	"github.com/obscuronet/obscuro-playground/go/obscuronode/enclave/core"
+	"github.com/obscuronet/obscuro-playground/go/obscuronode/nodecommon"
+)
+
+const expectedNonce = 777
+
+func TestCanDecryptRollup(t *testing.T) {
+	contractABI, err := abi.JSON(strings.NewReader(mgmtcontractlib.MgmtContractABI))
+	if err != nil {
+		panic(err)
+	}
+
+	rollupJSON, err := decryptRollup(generateEncryptedRollupHex(), contractABI)
+	if err != nil {
+		t.Fatalf("rollup decryption failed. Cause: %s", err)
+	}
+	var rollupJSONMap map[string]interface{}
+	err = json.Unmarshal(rollupJSON, &rollupJSONMap)
+	if err != nil {
+		t.Fatalf("rollup JSON unmarshalling failed. Cause: %s", err)
+	}
+
+	// We use the nonce as an indicator of whether the entire rollup was successfully decrypted.
+	recoveredNonce := rollupJSONMap["Header"].(map[string]interface{})["Nonce"].(float64)
+	if recoveredNonce != expectedNonce {
+		t.Fatalf("rollup JSON did not contain correct nonce")
+	}
+}
+
+func TestThrowsIfEncryptedRollupIsInvalid(t *testing.T) {
+	contractABI, err := abi.JSON(strings.NewReader(mgmtcontractlib.MgmtContractABI))
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = decryptRollup([]byte("invalid_rollup"), contractABI)
+	if err == nil {
+		t.Fatal("did not error on invalid rollup")
+	}
+}
+
+// Generates an encrypted rollup in hex encoding.
+func generateEncryptedRollupHex() []byte {
```

Comment:
You can use the data generator package here if it helps

Reply:
I'll skip that package, because I need to set specific values when creating the rollup so I can check they are present after decrypting.